server/bleep/src/webserver/repos.rs

```diff
@@ -144,7 +144,16 @@ pub(super) async fn delete_by_id(
         return Err(Error::new(ErrorKind::NotFound, "Can't find repository"));
     };
 
-    Ok(reporef.delete(&app).map(|_| json(ReposResponse::Deleted))?)
+    let deleted = match app.repo_pool.get_mut(&reporef) {
+        Some(mut result) => {
+            result.value_mut().mark_removed();
+            app.write_index().queue_sync_and_index(vec![reporef]);
+            Ok(())
+        }
+        None => Err(Error::new(ErrorKind::NotFound, "Repo not found")),
+    };
+
+    Ok(deleted.map(|_| json(ReposResponse::Deleted)))
```

Comment:
is this new code that needs review or has this been moved from elsewhere?

Reply:
This was a method on `RepoRef`. I've moved the logic into `webserver/repos.rs` to make consistent with other CRUD behaviour.

Reply:
I'd prefer if this was on RepoRef, to have a definitive interface for safe deletion of a repository, which is why I originally moved it there. Now we're binding this behaviour to the API, but there's actually no reason we wouldn't want to delete a repo through other mechanisms.

In general, I'd prefer if this kind of logic moved down from the API to lower layers in a safe wrapper, than having full-blown functions in-line in API handlers.

Reply:
While I agree that this logic shouldn't live at the API layer, we also don't want it split between the API and lower layers (as it is now). I propose that we move the deletion logic up for now, and then move all the index/delete logic down to a safe wrapper in a future PR (unless you want to add to this one right now).